gnomad/variant_qc/pipeline.py

```diff
@@ -320,3 +330,354 @@ def generate_sib_stats(
+    :param fp_to_tp: Ratio of FPs to TPs for creating the RF model. If set to 0, all training examples are used
+    :param num_trees: Number of trees in the RF model
+    :param max_depth: Maxmimum tree depth in the RF model
+    :param label_col: Name of column to store the training label
+    :param train_col: Name of column to store whether the site is used for training or not
+    :param test_intervals: The specified interval(s) will be held out for testing and used for evaluation only
+    :param filter_expr: Can be used to filter to specified intervals before training
+    :return: Table annotated with TP and FP training sets used in the RF training and the resulting RF model
+    """
+    ht = ht.annotate(tp=tp_expr, fp=fp_expr, _filter=filter_expr)
+
+    test_intervals_str = (
+        []
+        if not test_intervals
+        else [test_intervals]
+        if isinstance(test_intervals, str)
+        else test_intervals
+    )
+    test_intervals_locus = [
+        hl.parse_locus_interval(x, reference_genome=get_reference_genome(ht.locus).name)
+        for x in test_intervals_str
+    ]
+
+    if test_intervals_locus:
+        ht = ht.annotate_globals(test_intervals=test_intervals_locus)
+
+    ht = sample_rf_training_examples(
+        ht,
+        tp_col="tp",
+        fp_col="fp",
+        fp_to_tp=fp_to_tp,
+        label_col=label_col,
+        train_col=train_col,
+    )
+    ht = ht.persist()
+
+    rf_ht = ht
+
+    if filter_expr is not None:
+        logger.info("Filtering training set using filter_expr")
+        rf_ht = rf_ht.filter(rf_ht._filter)
+    rf_ht.drop("_filter")
+
+    summary = rf_ht.group_by("tp", "fp", train_col, label_col).aggregate(
+        n=hl.agg.count()
+    )
+    logger.info("Summary of TP/FP and RF training labels:")
+    summary.show(n=20)
+
+    rf_ht = rf_ht.filter(rf_ht[train_col])
+
+    logger.info(
+        "Training RF model:\nfeatures: {}\nnum_tree: {}\nmax_depth:{}\nTest intervals: {}".format(
+            ",".join(rf_features), num_trees, max_depth, ",".join(test_intervals_str),
+        )
+    )
+
+    rf_model = train_rf(
+        rf_ht,
+        features=rf_features,
+        label=label_col,
+        num_trees=num_trees,
+        max_depth=max_depth,
+    )
+
+    test_results = None
+    if test_intervals:
+        logger.info(f"Testing model on intervals {','.join(test_intervals_str)}")
+        test_ht = hl.filter_intervals(ht, test_intervals_locus, keep=True)
+        test_ht = test_ht.filter(hl.is_defined(test_ht[label_col]))
+        test_results = test_model(
+            test_ht, rf_model, features=rf_features, label=label_col
+        )
+
+    features_importance = get_features_importance(rf_model)
+    ht = ht.annotate_globals(
+        features_importance=features_importance,
+        features=rf_features,
+        test_intervals=test_intervals_str,
+        test_results=test_results,
+    )
+
+    return ht, rf_model
+
+
+def generate_final_rf_ht(
```

Comment:
So I think I wanted it to be RF-dependent since without that it seems like it is really just `add_filters_expr`. I changed the code to use `add_filters_expr` now. I am starting to wonder if I should just keep this out of `pipeline.py` and put back into just the UKBB files. I removed it for now, but we can talk about it more.